### PR TITLE
feat(abigen): descriptive deserialization errors

### PR DESCRIPTION
**Pull request by clifton**
currently abigen fails without a descriptive error msg

```
forge bind --module --bindings-path ./bindings/src/bindings --root ./contracts
[⠢] Compiling...
No files changed, compilation skipped
Error:
data did not match any variant of untagged enum JsonContract
```

this pr changes that to:

```
error parsing abi for contract: ExtRegistry.metadata
err: data did not match any variant of untagged enum JsonContract
abi: {
  "compiler": {
    "version": "0.8.15+commit.e14f2714"
  },
  "language": "Solidity",
  "output": {
    "abi": [
      {
        "inputs": [
          {
            "internalType": "contract GnosisSafe",
            "name": "_safe",
            "type": "address"
          }
.....
```

## PR Checklist

- [x] Updated the changelog

### CODE REVIEW

File: ethers-contract/ethers-contract-abigen/src/contract.rs [hunk 1]
```diff
@@ -191,7 +191,9 @@ impl Context {
         // holds the bytecode parsed from the abi_str, if present
         let mut contract_bytecode = None;
 
-        let (abi, human_readable, abi_parser) = parse_abi(&abi_str)?;
+        let (abi, human_readable, abi_parser) = parse_abi(&abi_str).map_err(|err| {
```
Comment (mattsse) [inline ethers-contract/ethers-contract-abigen/src/contract.rs:194]: let's use `wrap_err`

**Review by mattsse — CHANGES_REQUESTED** [review]
let's use `wrap_err` so we keep the original error context

File: ethers-contract/ethers-contract-abigen/src/contract.rs [hunk 2]
```diff
@@ -351,7 +353,8 @@ fn parse_abi(abi_str: &str) -> Result<(Abi, bool, AbiParser)> {
         (abi, true, abi_parser)
     } else {
         // a best-effort coercion of an ABI or an artifact JSON into an artifact JSON.
-        let contract: JsonContract = serde_json::from_str(abi_str)?;
+        let contract: JsonContract = serde_json::from_str(abi_str)
+            .wrap_err_with(|| eyre::eyre!("failed deserializing abi:\n{}", abi_str))?;
```
Comment (clifton) [inline ethers-contract/ethers-contract-abigen/src/contract.rs:357]: is this better @mattsse

**Review by mattsse — APPROVED** [review]
thanks, this is much more useful

**Pull request merged**